subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/resolver/ResolverEventLogger.kt

```diff
@@ -76,7 +76,10 @@ object DefaultResolverEventLogger : ResolverEventLogger {
 
     private
     val outputFile by lazy {
-        File(outputDir(), "resolver-${timestampForFileName()}.log")
+        outputDir().let { logDir ->
+            cleanupLogDirectory(logDir)
```

Comment:
❓ That would happen at least once per edited script type per project session, right?

I wonder if we shouldn't impose a harder limit to the amount of directory scanning here and say, cleanup once a week, for instance. What do you think?

Reply:
Once a day like [Gradle project caches cleanup](https://docs.gradle.org/current/userguide/directory_layout.html#dir:project_root:cache_cleanup) sounds good.

Reply:
I think this should do, PTAL